core/src/test/scala/unit/kafka/server/ListenerAndPrincipalBuilderTest.scala

```diff
@@ -0,0 +1,71 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package kafka.server
+
+import java.net.Socket
+import java.util.Properties
+
+import kafka.utils.TestUtils
+import org.apache.kafka.common.network.ListenerName
+import org.apache.kafka.common.protocol.types.Struct
+import org.apache.kafka.common.requests._
+import org.apache.kafka.common.security.auth.{AuthenticationContext, KafkaPrincipal, KafkaPrincipalBuilder}
+import org.junit.Assert._
+import org.junit.{Before, Test}
+
+class ListenerAndPrincipalBuilderTest extends BaseRequestTest {
```

Comment:
Do we need a new test for this? These tests are expensive since they involve starting several brokers. Ideally, we'd update one of the existing tests to also check the listener name.

Reply:
We had a look at other tests extending `BaseRequestTest` and this does not really fit anywhere. We tried merging it into `KafkaMetricReporterExceptionHandlingTest` (one test checking both logics) but it really looked awful. As each test needs its own setup, the overall logic made little sense even with comments.

Yes the test takes 5 secs but we think it's clearer to keep it separated.

Reply:
What about `PlaintextEndToEndAuthorizationTest`?